### ISSUE TITLE
The program happy is required but it could not be found

### ISSUE THREAD

**Issue by andrewthad** [haskell/actions#186]
Started getting a build failures recently from a GitHub action that had previously been working. I think it has something to do with this action moving from cabal 3.6 to 3.8. Here's the action I use:

```
name: build-and-test
on: [push]
jobs:
  all:
    runs-on: ubuntu-20.04
    steps:
      - uses: actions/checkout@v3
      - run: ./scripts/hooks/commit-hook.bash
      - run: |
               sudo apt-get update
               sudo apt-get install libsystemd-dev
      - run: .github/workflows/kafka-install.bash
      - uses: haskell/actions/setup@v2
        with:
          ghc-version: 9.2.5
      - run: cabal build
      - run: cabal test
      - run: ./integration/test.sh
```

The `cabal build` step fails with

```
Failed to build exe:happy from happy-1.20.1.
Build log (
/home/runner/.cabal/logs/ghc-9.2.5/happy-1.20.1-e-happy-5b2511589bcadeac57971eeaead82f5eb2bd857b0f89940af650bc691c95e13d.log
):
Configuring executable 'happy' for happy-1.20.1..
Error: cabal: Failed to build exe:happy from happy-1.20.1 (which is required
by exe:allsight-web from allsight-3.0.16 and exe:allsight from
allsight-3.0.16). See the build log above for details.

Preprocessing executable 'happy' for happy-1.20.1..
Error: cabal-3.8.1.0: The program 'happy' is required but it could not be
found
```

When I use cabal 3.8 locally on my laptop, I get this same problem. It seems like cabal 3.6 had a way of installing happy for the user, and cabal 3.8 does not. I am able to resolve the issue locally by installing `happy` through other means and putting it on the `PATH`. I don't know how to do this same thing in an action. I'm not sure if is an issue with `haskell/actions`, ghcup, or cabal.

**Comment by andreasabel** [comment]
Sorry, this is because I released a broken `happy-1.20.1`.  However, this should be fixed now, since this version is deprecated on Hackage.  
Can you try to rerun your action?
- https://github.com/haskell/happy/issues/250

**Comment by andreasabel** [comment]
I released `happy-1.20.1.1` fixing the core problem, so I believe this issue is fixed as well.
Closing, but reopen if problem persists!
